### ISSUE TITLE
question: how much this is stable?

### ISSUE THREAD

**Issue by kintaro1981** [TwinFuture/virtualmin-nginx-ssl#1]
I'm asking this because at the moment I'm using this:
https://github.com/Real-Gecko/virtualmin-nginx-fpm/
but it have some issues.

is this stable?
can you upload a little of documentation or simply some more info about it?

**Comment by TwinFuture** [comment]
Hey and sorry for the late response!
I believe this is stable, I have done a lot of debugging and trial and error to ensure its working as it should work. I looked at the mod you posted, its very basic and looks like it would have a lot of problems as a lot of functions/features simply do not exist in the mod you posted. 

I guess I am not very good at writing documentation. I will improve on this and outline what it does and how to use it more as soon as I get some more free time.

You should test it out, let me know if there is any problems or additions you feel would be necessary.
I feel it is very stable and working as it should work.. It could do with some features added like I do plan on adding a GUI panel for modifying PHP-FPM for every virtual host.,

**Comment by kintaro1981** [comment]
Thank you for your reply. 😄 

What do you think I have to do to migrate from the Real-Gecko's to your module?

your module will read the nginx and php-fpm configuration as it is or it is going to look in particular directories?

**Comment by TwinFuture** [comment]
Hey to be honest, I have not added modules when virtual hosts have already been created. But I believe it would be relatively simple and should pickup your current virtual host files you have installed with nginx.

If it doesn't automatically discover your virtual hosts, its probably because a file within etc->webmin->virtualmin-nginx is missing.

Which should be named "domain-name".acl Replace domain name with your hosts domain name that you used to create the virtual server.

The contents of the file should look like this

```
vhosts=domain-name.com
stop=0
global=0
user=user
root=/home/domain
edit=0
logs=0
```

Don't add the public_html part to the root just the home dir for the domain.
So basically if your host file is not recognised then you can simply just add this file in that location and it should be picked up from there.

Backup your old MOD, then if this one doesn't do as you please, you can simply restore the old mod files..

Other than that everything should be okay.. Any problems give us a shout!

Thanks.

**Comment by 7starsone** [comment]
if that's it... it's enough to create a script which reads the current managed domain names and user directories...and auto writes those files with the proper content inside the directory

**Comment by 7starsone** [comment]
and then... you can set options eg. 1) Use Nginx for all your domains? Yes, No 2) Use Nginx for domains listed (with textfield where you insert your domains or checkboxes which read all the domains and you can select what you want...) this is a way to do it...

**Comment by TwinFuture** [comment]
@7starsone Yeah that's a great idea, but the previous mod he had installed also should have an ACL, which is what it says it is... This allows the user of the domain to modify their nginx configuration for that domain name if the controls are enabled within virtualmin. Stops someone who has an account in virtualmin from logging in and modifying someone else nginx hosts settings. I will need to look into this. To see exactly what could be missing if using another mod. Currently I only added it to make new virtual hosts and restore/backup hosts. But I do believe everything should transition over properly.
Thanks for the input!
